airflow/models.py

```diff
@@ -3732,6 +3732,37 @@ def latest_execution_date(self, session=None):
         ).scalar()
         return execution_date
 
+    @property
+    def next_run_date(self):
+        """
+        Returns the next run date for which the dag will be scheduled
+        """
+        next_run_date = None
+        if not self.latest_execution_date:
+            # First run
+            task_start_dates = [t.start_date for t in self.tasks]
```

Comment:
I'm not sure if this function will make sense when `latest_execution_date` is not available.

Understand you're trying to decide the next run using `start_date`. But `catchup` of the DAG can be either `True` or `False`, and it will affect the actual next execution date (ref: https://airflow.apache.org/scheduler.html?highlight=backfill#backfill-and-catchup).